### PR TITLE
feat: map "if_exists" value to LoadJobConfig.WriteDisposition

### LINKED ISSUE #118: CLN: Use WriteDisposition for if_exists=fail/replace/append behavior in to_gbq
Now that to_gbq is using load jobs instead of streaming to write to table, it could use the write disposition parameter to perform this behavior.

I believe by doing so we would unlock https://github.com/pydata/pandas-gbq/issues/43 since using write disposition could replace just one partition, but the current logic of deleting and recreating a table cannot handle partitions.

### PR DESCRIPTION
**Pull request by aribray**
Use `LoadJobConfig.WriteDisposition` to replace `if_exists='fail'/'replace'/'append'` behavior in `to_gbq()`

The WriteDisposition value specifies the action that occurs if destination table already exists.

- if_exists="fail" ----> write_disposition="WRITE_EMPTY" 
- if_exists="append" ----> write_disposition="WRITE_APPEND"
- if_exists="replace" ----> write_disposition="WRITE_TRUNCATE"

By using WriteDisposition, we can move away from the `delete_and_recreate_table()` behavior

Fixes #118 #43 #453

### CODE REVIEW

File: pandas_gbq/load.py [hunk 1]
```diff
@@ -113,20 +113,19 @@ def load_parquet(
...
     dataframe: pandas.DataFrame,
     destination_table_ref: bigquery.TableReference,
+    write_disposition: str,
     location: Optional[str],
     schema: Optional[Dict[str, Any]],
     billing_project: Optional[str] = None,
 ):
     job_config = bigquery.LoadJobConfig()
-    job_config.write_disposition = "WRITE_APPEND"
-    job_config.create_disposition = "CREATE_NEVER"
```
Comment (aribray) [inline pandas_gbq/load.py:122]: The `create_disposition` attribute defaults to `CREATE_IF_NEEDED`

File: ci/requirements-3.7-0.24.2.conda [hunk 2]
```diff
@@ -1,14 +1,17 @@
 codecov
```
Comment (aribray) [inline ci/requirements-3.7-0.24.2.conda:1]: Pinning dependencies was causing a bunch of problems for the CircleCI (3.7) build. We should move these tests over to Kokoro

**Comment by aribray** [comment]
@tswast are the version bumps/PR description added to the Changelog automatically?

**Comment by tswast** [comment]
> @tswast are the version bumps/PR description added to the Changelog automatically?

No, that needs to be included in the commit message when we 'squash and merge'. See: https://github.com/googleapis/release-please#what-if-my-pr-contains-multiple-fixes-or-features

File: pandas_gbq/load.py [hunk 3]
```diff
@@ -234,13 +236,15 @@ def load_chunks(
     schema=None,
     location=None,
     api_method="load_parquet",
+    write_disposition="WRITE_EMPTY",
```
Comment (rosiezou) [inline pandas_gbq/load.py:239]: nit: define a global string constant for this arg, same with `api_method`

File: pandas_gbq/gbq.py [hunk 4]
```diff
@@ -1114,15 +1109,21 @@ def to_gbq(
...
-    if if_exists not in ("fail", "replace", "append"):
-        raise ValueError("'{0}' is not valid for if_exists".format(if_exists))
-
     if "." not in destination_table:
         raise NotFoundException(
             "Invalid Table Name. Should be of the form 'datasetId.tableId' or "
             "'projectId.datasetId.tableId'"
         )
 
+    if if_exists not in ("fail", "replace", "append"):
```
Comment (shollyman) [inline pandas_gbq/gbq.py:1118]: minor consideration:  Should we case coerce `if_exists` to lowercase when evaluating, or throw the value error unless they're exact match?  I'm fine with the strictness approach, but something to consider.

**Review by shollyman — APPROVED** [review]
Thanks for putting this together.  Switching to leveraging write disposition cleans things up nicely.

File: pandas_gbq/gbq.py (hunk 4 above)
Comment (aribray) [reply pandas_gbq/gbq.py:1118]: That's a good point. I'll make a note to do that

**Pull request merged**